rules/azurerm_app_service_app_insights_hidden_link.go

```diff
@@ -0,0 +1,229 @@
+		},
+		Blocks: []hclext.BlockSchema{
+			{
+				Type: "lifecycle",
+				Body: &hclext.BodySchema{
+					Attributes: []hclext.AttributeSchema{
+						{Name: appServiceIgnoreChangesAttrName},
+					},
+				},
+			},
+			{
+				Type: "site_config",
+				Body: &hclext.BodySchema{
+					Attributes: []hclext.AttributeSchema{
+						{Name: appServiceSiteConfigAppInsightsConnectionKey},
+						{Name: appServiceSiteConfigAppInsightsKey},
+					},
+				},
+			},
+		},
+	}, nil)
+
+	if err != nil {
+		return err
+	}
+
+	for _, resource := range resources.Blocks {
+		logger.Debug("checking", "resource type", resource.Labels[0], "resource name", resource.Labels[1])
+
+		// Check if Application Insights is configured
+		hasAppInsights := false
+
+		// Check in app_settings
+		if appSettingsAttr, exists := resource.Body.Attributes[appServiceAppSettingsAttrName]; exists {
+			err := runner.EvaluateExpr(appSettingsAttr.Expr, func(val map[string]string) error {
+				for key := range val {
+					if strings.EqualFold(key, appServiceAppInsightsConnectionKey) || strings.EqualFold(key, appServiceAppInsightsInstrumentKey) {
+						hasAppInsights = true
+						break
+					}
+				}
+				return nil
+			}, nil)
+
+			if err != nil {
+				return err
+			}
+		}
+
+		// Check in site_config block
+		if !hasAppInsights {
+			for _, block := range resource.Body.Blocks {
+				if block.Type == "site_config" {
+					// Check for application_insights_connection_string
+					if attr, exists := block.Body.Attributes[appServiceSiteConfigAppInsightsConnectionKey]; exists {
+						err := runner.EvaluateExpr(attr.Expr, func(val string) error {
+							if val != "" {
+								hasAppInsights = true
+							}
+							return nil
+						}, nil)
+						if err != nil {
+							return err
+						}
+					}
+
+					// Check for application_insights_key
+					if !hasAppInsights {
+						if attr, exists := block.Body.Attributes[appServiceSiteConfigAppInsightsKey]; exists {
+							err := runner.EvaluateExpr(attr.Expr, func(val string) error {
+								if val != "" {
+									hasAppInsights = true
+								}
+								return nil
+							}, nil)
+							if err != nil {
+								return err
+							}
+						}
+					}
+					break
+				}
+			}
+		}
+
+		// If Application Insights is not configured, skip this resource
+		if !hasAppInsights {
+			logger.Debug("no Application Insights configuration found", "resource type", resource.Labels[0], "resource name", resource.Labels[1])
+			continue
+		}
+
+		// Check if lifecycle block exists and contains proper ignore_changes
+		hasProperIgnoreChanges := false
+
+		for _, block := range resource.Body.Blocks {
+			if block.Type == "lifecycle" {
+				if ignoreChangesAttr, ok := block.Body.Attributes[appServiceIgnoreChangesAttrName]; ok {
+					// Parse the ignore_changes expression to find ignored tag keys
+					foundTags := 0
+					tagsIgnored := false
+					if exprs, diags := hcl.ExprList(ignoreChangesAttr.Expr); diags == nil {
```

Comment:
We should use `diags.HasErrors()` instrad of `diags == nil`.

```suggestion
					if exprs, diags := hcl.ExprList(ignoreChangesAttr.Expr); !diags.HasErrors() {
```

https://github.com/hashicorp/hcl/blob/v2.24.0/diagnostic.go#L142-L151